### PR TITLE
harbfuzz: add new version 10.1.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **harfbuzz/10.1.0**

https://github.com/harfbuzz/harfbuzz/compare/8.3.1...10.1.0

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/harfbuzz/all/conanfile.py [hunk 1]
```diff
@@ -154,14 +154,21 @@ def is_vs_2017():
...
         tc.cpp_args += cxxflags
         tc.generate()
 
     def build(self):
         apply_conandata_patches(self)
-        replace_in_file(self, os.path.join(self.source_folder, "meson.build"), "subdir('util')", "")
+        if Version(self.version) < "7.3.0":
+            replace_in_file(self, os.path.join(self.source_folder, "meson.build"), "subdir('util')", "")
+        if Version(self.version) >= "8.4.0":
+            replace_in_file(self, os.path.join(self.source_folder, "meson.build"), "version: freetype_min_version,", "")
```
Comment (jwillikers) [inline recipes/harfbuzz/all/conanfile.py:171]: This workaround shouldn't be necessary when #22974 is merged.

**Review by jwillikers — APPROVED** [review]

File: recipes/harfbuzz/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/harfbuzz/all/conanfile.py:171]: @jwillikers 
Thanks a lot!
I'm waiting for it.

Comment (RubenRBS) [reply recipes/harfbuzz/all/conanfile.py:171]: Is this no longer needed then?

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

File: recipes/harfbuzz/all/conanfile.py (hunk 1 above)
Comment (AbrilRBS) [reply recipes/harfbuzz/all/conanfile.py:171]: @toge bump here - can this workaround be removed now that the linked PR has been merged?

Comment (perseoGI) [reply recipes/harfbuzz/all/conanfile.py:171]: Ping @toge

**Pull request closed without merging**